src/python-common/ceph/rgw/rgwam_core.py

```diff
@@ -193,46 +190,54 @@ def __init__(self, env: EnvArgs):
         self.env = env
 
     def list(self):
-        ze = ZoneEnv(self.env)
-
-        params = ['realm',
-                  'list']
-
-        return RGWAdminJSONCmd(ze).run(params)
+        try:
+            ze = ZoneEnv(self.env)
+            params = ['realm', 'list']
+            output = RGWAdminJSONCmd(ze).run(params)
+            return output.get('realms') or []
+        except RGWAMException:
+            # in case the realm list is empty an exception is raised
```

Comment:
@adk3798 not sure if there other conditions but if no realms exist we get an exception and an error as the following:

```
[ceph: root@ceph-node-0 /]# radosgw-admin realm list
failed to list realms: (2) No such file or directory
[ceph: root@ceph-node-0 /]# echo $?
2
```
This is something that rgw does (IMHO we shouldn't fail to list realms when the list is empty). Anyway, I'll add a log with the full exception just in case.